internal/definition/awsintegration/schema_integration.go

```diff
@@ -0,0 +1,312 @@
+					},
+					"metric": {
+						Type:        schema.TypeString,
+						Required:    true,
+						Description: "AWS metric that you want to pick statistics for",
+					},
+					"stats": {
+						Type:        schema.TypeSet,
+						Required:    true,
+						Description: "AWS statistics you want to collect",
+						Elem: &schema.Schema{
+							Type: schema.TypeString,
+						},
+					},
+				},
+			},
+		},
+		"sync_custom_namespaces_only": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     false,
+			Description: "Indicates that Splunk Observability should sync metrics and metadata from custom AWS namespaces only (see the `custom_namespace_sync_rule` field for details). Defaults to `false`.",
+		},
+		"collect_only_recommended_stats": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     false,
+			Description: "Indicates that Splunk Observability should only sync recommended statistics",
+		},
+	}
+}
+
+func decodeTerraform(rd *schema.ResourceData) (*integration.AwsCloudWatchIntegration, error) {
+	cwi := &integration.AwsCloudWatchIntegration{
+		Type:                        "AWSCloudWatch",
+		Id:                          rd.Get("integration_id").(string),
+		Name:                        rd.Get("name").(string),
+		Enabled:                     rd.Get("enabled").(bool),
+		EnableAwsUsage:              rd.Get("enable_aws_usage").(bool),
+		ImportCloudWatch:            rd.Get("import_cloud_watch").(bool),
+		EnableCheckLargeVolume:      rd.Get("enable_check_large_volume").(bool),
+		SyncCustomNamespacesOnly:    rd.Get("sync_custom_namespaces_only").(bool),
+		CollectOnlyRecommendedStats: rd.Get("collect_only_recommended_stats").(bool),
+		Regions:                     convert.SchemaListAll(rd.Get("regions"), convert.ToString),
+		Services:                    convert.SchemaListAll(rd.Get("services"), convert.ToStringLike[integration.AwsService]),
+		NamespaceSyncRules:          convert.SchemaListAll(rd.Get("namespace_sync_rule"), convert.ToAWSNamespaceRule),
+		CustomNamespaceSyncRules:    convert.SchemaListAll(rd.Get("custom_namespace_sync_rule"), convert.ToAWSCustomNamespaceRule),
+		CustomCloudWatchNamespaces:  strings.Join(convert.SchemaListAll(rd.Get("custom_cloudwatch_namespaces"), convert.ToString), ","),
+	}
+
+	if v, ok := rd.GetOk("use_metric_streams_sync"); ok && v.(bool) {
+		cwi.MetricStreamsSyncState = "ENABLED"
+	} else if rd.HasChange("use_metric_streams_sync") {
+		// use_metric_streams_sync is false, and it has changed, meaning it was ENABLED before
+		cwi.MetricStreamsSyncState = "CANCELING"
+	}
+
+	if v, ok := rd.GetOk("enable_log_sync"); ok && v.(bool) {
+		cwi.LogsSyncState = "ENABLED"
+	} else if rd.HasChange("enable_log_sync") {
+		// enable_logs_sync is false, and it has changed, meaning it was ENABLED before
+		cwi.LogsSyncState = "CANCELING"
+	}
+
+	if v, ok := rd.GetOk("external_id"); ok && v != "" {
```

Comment:
I am not sure about this as on the API level `externalId` is always defined. Is it `nil` when `signalfx_aws_token_integration` resource is used? 

Shouldn't this logic depend simply on the `authMethod` field?

Reply:
That is a great point, let me clean that up.

Reply:
>  Is it nil when signalfx_aws_token_integration resource is used?
Yes, if you have created the details for token integration. 

> Shouldn't this logic depend simply on the authMethod field?

So interestingly, this is currently a computed field meaning it isn't possible to be set by the user and so the provider figures it out.

Reply:
TBH I would rather have this logic to depend on the `authMethod` if possible to avoid some regression in the future should the `externalId` become always defined (after all this is how it looks like on the API level).

Reply:
Alright, I have updated the code to require users to set AuthMethod as part of their terraform config, and provide validation on it.